crates/matrix-sdk/src/room_directory_search.rs

```diff
@@ -0,0 +1,370 @@
+// Copyright 2024 Mauro Romito
+// Copyright 2024 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Types for searching the public room directory.
+
+use eyeball_im::{ObservableVector, VectorDiff};
+use futures_core::Stream;
+use imbl::Vector;
+use ruma::{
+    api::client::directory::get_public_rooms_filtered::v3::Request as PublicRoomsFilterRequest,
+    directory::{Filter, PublicRoomJoinRule},
+    OwnedMxcUri, OwnedRoomAliasId, OwnedRoomId,
+};
+
+use crate::{Client, Result};
+
+/// This struct represents a single result of a room directory search.
```

Comment:
```suggestion
/// This struct represents a single result of a room directory search.
///
/// It's produced by [`RoomDirectorySearch::results`].
```

I believe it's important to link/connect types together, it's easier for new developers.